packages/backend/src/Application.ts

```diff
@@ -0,0 +1,45 @@
+import { Logger } from '@l2beat/backend-tools'
+
+import { ApiServer } from './api/ApiServer'
+import { Config } from './config'
+import { ApplicationModule } from './modules/ApplicationModule'
+import { createHealthModule } from './modules/HealthModule'
+import { Database } from './peripherals/database/shared/Database'
+
+// TODO: Error reporting
+export class Application {
+  start: () => Promise<void>
+
+  constructor(config: Config) {
+    const loggerOptions = { ...config.logger }
+
+    const logger = new Logger(loggerOptions)
+
+    const database = new Database(config.database.connection, logger)
+
+    const modules: (ApplicationModule | undefined)[] = [
+      createHealthModule(config),
+    ]
+
+    const apiServer = new ApiServer(
+      config.api.port,
+      logger,
+      modules.flatMap((x) => x?.routers ?? []),
+    )
+
+    this.start = async () => {
+      logger.for(this).info('Starting')
+
+      await apiServer.listen()
+
+      if (config.database.freshStart) {
+        await database.rollbackAll()
+      }
+      await database.migrateToLatest()
+
+      for (const module of modules) {
+        await module?.start?.()
+      }
```

Comment:
```suggestion
      logger.for(this).info('Starting')

      if (config.database.freshStart) {
        await database.rollbackAll()
      }
      await database.migrateToLatest()

      for (const module of modules) {
        await module?.start?.()
      }
      
      await apiServer.listen()
```

we should probably boot up socket after migrations

Reply:
Also it would remove logic responsible for waiting for migrations before booting up knex connection - nit pick tho - can be done separately